### PR TITLE
Include Summer23 branch in Latinos 

### PR DESCRIPTION
**Pull request by BlancoFS**
**Summer23**

Given the changes and fixes to commonly used parts of the code, we think it is better to merge this code into the master branch instead of opening another one. Now we enumerate the features of the code that have been implemented/modified:

- Fix to searchFiles (There was a bug so that the first file was always missing)
- Fix to Snapshot    (The copyFromInputFiles function needs to apply CloneTree(). A memory corruption occurred when reading output files)
- Fix to LeptonSel module  (Fix to DeltaR2 between CleanJets and Leptons)
- Fix to mRDF   (Avoid error from redefinition of "CUT" + add Histo1D function)
- Include Sites_cfg.py + EosTmpDir for postprocessing    (As a user you can set your own paths there)
- Include a script to loop over jobs and files to check if someone has failed. "doCheck.py" in Tools.
- Include PU reweight module + PUWeight_cfg.py
- Include TriggerSFMaker + TrigMaker_cfg.py + Efficiencies for Run2022FG and UL
- Include Golden JSons for Run2022
- Include LeptonSF module
- Include Muon + Electron SF for Run2022FG
- Update Production_cfg.py, Steps_cfg. Summer22EE and Run2022FG configuration files
- Include formulasToAdd for MC and Data for 2022EE production
- Include l2Kin, l3Kin, and l4Kin (From Nicolo's PR and Susan's PR respectively)

In summary, this is a fully working version that includes most of the necessary modules to run the Run2022FG production

### CODE REVIEW

**Comment by NTrevisani** [comment]
I am merging the PR to a new branch called Run3, to ensure it does not affect the current UL branches and can develop independently.
